service/controller/v18/resource/nodeindexstatus/create_test.go

```diff
@@ -0,0 +1,401 @@
+package nodeindexstatus
+
+import (
+	"context"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/google/go-cmp/cmp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_EnsureCreated(t *testing.T) {
+	testCases := []struct {
+		name              string
+		kvmConfig         *v1alpha1.KVMConfig
+		expectedKVMConfig *v1alpha1.KVMConfig
+		errorMatcher      func(error) bool
+	}{
+		{
+			name: "case 0: no status, add indexes for nodes",
+			kvmConfig: &v1alpha1.KVMConfig{
+				Spec: v1alpha1.KVMConfigSpec{
+					Cluster: v1alpha1.Cluster{
+						Masters: []v1alpha1.ClusterNode{
+							{
+								ID: "1dkfa",
+							},
+						},
+						Workers: []v1alpha1.ClusterNode{
+							{
+								ID: "1ddqd",
+							},
+							{
+								ID: "1dcli",
+							},
+							{
+								ID: "1dlev",
+							},
+						},
+					},
+				},
+			},
+			expectedKVMConfig: &v1alpha1.KVMConfig{
+				Spec: v1alpha1.KVMConfigSpec{
+					Cluster: v1alpha1.Cluster{
+						Masters: []v1alpha1.ClusterNode{
+							{
+								ID: "1dkfa",
+							},
+						},
+						Workers: []v1alpha1.ClusterNode{
+							{
+								ID: "1ddqd",
+							},
+							{
+								ID: "1dcli",
+							},
+							{
+								ID: "1dlev",
+							},
+						},
+					},
+				},
+				Status: v1alpha1.KVMConfigStatus{
+					KVM: v1alpha1.KVMConfigStatusKVM{
+						NodeIndexes: map[string]int{
+							"1dkfa": 1,
+							"1ddqd": 2,
+							"1dcli": 3,
+							"1dlev": 4,
+						},
+					},
+				},
+			},
+			errorMatcher: nil,
+		},
+		{
+			name: "case 1: existing nodeindexes, add new node node",
+			kvmConfig: &v1alpha1.KVMConfig{
+				Spec: v1alpha1.KVMConfigSpec{
+					Cluster: v1alpha1.Cluster{
+						Masters: []v1alpha1.ClusterNode{
+							{
+								ID: "1dkfa",
+							},
+						},
+						Workers: []v1alpha1.ClusterNode{
+							{
+								ID: "1ddqd",
+							},
+							{
+								ID: "1dcli",
+							},
+							{
+								ID: "1dlev",
+							},
+							{
+								ID: "1dmus",
+							},
+						},
+					},
+				},
+				Status: v1alpha1.KVMConfigStatus{
+					KVM: v1alpha1.KVMConfigStatusKVM{
+						NodeIndexes: map[string]int{
+							"1dkfa": 1,
+							"1ddqd": 2,
+							"1dcli": 3,
+							"1dlev": 4,
```

Comment:
Can we have single letter IDs or single letter differences in them? It is extremely hard for me to read this.

Reply:
Fixed